pages/events.js

```diff
@@ -19,15 +19,36 @@ export async function getServerSideProps(context) {
   };
 }
 
+const groupEvents = (events) => {
+  const groups = {}
+  events.forEach(event => {
```

Comment:
I think this could be done with a `reduce`, but even better with an aggregation on the API in mongo https://www.mongodb.com/docs/manual/reference/operator/aggregation/group/